pkg/wpa/passphrase/passphrase.go

```diff
@@ -0,0 +1,36 @@
+package passphrase
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+
+	"golang.org/x/crypto/pbkdf2"
+)
+
+const ResultFormat = `network={
+	ssid="%s"
+	#psk="%s"
+	psk=%s
+}
+`
+
+func errorCheck(essid string, pass string) error {
+	if len(pass) < 8 || len(pass) > 63 {
+		return fmt.Errorf("Passphrase must be 8..63 characters")
+	}
+	if len(essid) == 0 {
+		return fmt.Errorf("essid cannot be empty")
+	}
+	return nil
+}
+
+func Run(essid string, pass string) ([]byte, error) {
+	if err := errorCheck(essid, pass); err != nil {
+		return nil, err
+	}
+
+	pskBinary := pbkdf2.Key([]byte(pass), []byte(essid), 4096, 32, sha1.New)
```

Comment:
Should these integers be constants (4096, 32) ?

Reply:
Hmm, I was following this source code http://docs.ros.org/diamondback/api/wpa_supplicant/html/wpa__passphrase_8c_source.html